### PR TITLE
rename datadogagentdeployment to datadogagent

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

rename datadogagentdeployment to datadogagent

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/apis/datadoghq/v1alpha1/const.go [hunk 1]
```diff
@@ -6,7 +6,7 @@
 package v1alpha1
 
 const (
-	// AgentDeploymentNameLabelKey label key use to link a Resource to a DatadogAgentDeployment
+	// AgentDeploymentNameLabelKey label key use to link a Resource to a DatadogAgent
 	AgentDeploymentNameLabelKey = "agentdeployment.datadoghq.com/name"
```
Comment (ahmed-mez) [inline pkg/apis/datadoghq/v1alpha1/const.go]: should we consider renaming these as well? `agent.datadoghq.com/name`

File: pkg/apis/datadoghq/v1alpha1/datadogagent_types.go [hunk 2]
```diff
@@ -533,45 +533,45 @@ type ImageConfig struct {
 	PullSecrets *[]corev1.LocalObjectReference `json:"pullSecrets,omitempty"`
 }
 
-// DatadogAgentDeploymentState type representing the ClusterAgent as a DaemonSet deployment state
-type DatadogAgentDeploymentState string
+// DatadogAgentState type representing the ClusterAgent as a DaemonSet deployment state
```
Comment (ahmed-mez) [inline pkg/apis/datadoghq/v1alpha1/datadogagent_types.go]:
```suggestion
// DatadogAgentState type representing the Agent as a DaemonSet deployment state
```

File: pkg/apis/datadoghq/v1alpha1/datadogagent_types.go [hunk 3]
```diff
@@ -533,45 +533,45 @@ type ImageConfig struct {
...
 
-	// Conditions Represents the latest available observations of a DatadogAgentDeployment's current state.
+	// Conditions Represents the latest available observations of a DatadogAgent's current state.
 	// +listType=set
-	Conditions []DatadogAgentDeploymentCondition `json:"conditions,omitempty"`
+	Conditions []DatadogAgentCondition `json:"conditions,omitempty"`
 }
 
-// DatadogAgentDeploymentAgentStatus defines the observed state of Agent running as DaemonSet
+// DatadogAgentAgentStatus defines the observed state of Agent running as DaemonSet
```
Comment (ahmed-mez) [inline pkg/apis/datadoghq/v1alpha1/datadogagent_types.go]:
```suggestion
// DatadogAgentDaemonSetStatus defines the observed state of Agent running as DaemonSet
```

File: pkg/apis/datadoghq/v1alpha1/datadogagent_types.go [hunk 4]
```diff
@@ -533,45 +533,45 @@ type ImageConfig struct {
...
 // +k8s:openapi-gen=true
-type DatadogAgentDeploymentAgentStatus struct {
+type DatadogAgentAgentStatus struct {
```
Comment (ahmed-mez) [inline pkg/apis/datadoghq/v1alpha1/datadogagent_types.go]:
```suggestion
type DatadogAgentDaemonSetStatus struct {
```

File: examples/datadog-agent-all.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog-agent
+spec:
+  credentials:
+    apiKey: <appkey>
```
Comment (ahmed-mez) [inline examples/datadog-agent-all.yaml]:
```suggestion
    apiKey: <apikey>
    appKey: <appkey>
```

File: pkg/apis/datadoghq/v1alpha1/const.go (hunk 1 above)
Comment (clamoriniere) [reply pkg/apis/datadoghq/v1alpha1/const.go]: good idea

**Pull request merged**